n4js-libs/packages/n4js-cli/src/n4js/JreProvider.n4js

```diff
@@ -96,40 +101,44 @@ export public async function ensureJRE(log: {function(string)} = npmlog.info) :
     }
 
     // Download and extract JRE
-    const jreInfo = getJreInfo(log);
+    const jreInfo = getJreInfo();
     if (jreInfo) {
-        log("Downloading and extracting JRE into local folder.");
-
-        await downloadJRE(jreInfo, log);
-        await extractJRE(jreInfo, log);
+        if (isPlatformProblemMuslLibc()) {
+            log("Detected musl libc on this platform: Will not download Adopt OpenJDK since musl libc is not supported by Adopt OpenJDK.");
+        } else {
+            log("Downloading and extracting JRE (Adopt OpenJDK) into local folder.");
+            
+            try {
+                await downloadJRE(jreInfo, log);
+                await extractJRE(jreInfo, log);
+                return;
+            } catch (err) {
+                log(err as string);
+            }
+        }
 
     } else {
         log("Platform not supported: " + process.platform);
-        log("Please install JRE in version " + JRE_VERSION + " and make the call to 'bin/java' globally available.");
     }
+    log("Please install JRE in version " + JRE_VERSION + " and make the call to 'bin/java' globally available.");
 }
 
-function getJreInfo(log: {function(string)}) : JreInfo {
+function getJreInfo() : JreInfo {
     const platform = process.arch + "_" + process.platform;
-    if (platform == Platform.x64_linux && isMuslLibc()) {
-        log("Musl libc is not supported by Adopt OpenJDK");
-        return null;
-    }
     return JRE_INFO_MAP.get(platform as Platform);
 }
 
 
 /**
  * Downloads the AdoptOpenJDK JRE.
  */
-async function downloadJRE(jreInfo: JreInfo, log: {function(string)}) : Promise<any, any> {
+async function downloadJRE(jreInfo: JreInfo, log: {function(string)}) : void {
```

Comment:
consider changing the return type of `downloadFile` accordingly (for consistency)